Added a decorator that checks the origin header of a websocket.connec…

dhepper opened the pull request:
…t message. This partially fixes #585

I created a new module at the top level because I wasn't sure where to put it. It should probably be moved to a more appropriate location.

For a working example, check out https://github.com/dhepper/channels-examples/tree/origin_check

andrewgodwin commented:
I like the design of this, it's clean and looks good! Two things:

* Let's call it `channels/security/websocket.py`, which fits more with how we try to separate things a little.
* Are you able to include basic docs in the PR? Not everyone is happy writing docs, and if you can't, I can do it, it'll just take a bit longer :)

dhepper commented:
Thanks for your quick feedback!

* I've renamed the new module to `channels/security/websockets.py` (with a trailing s to match the existing modules)
* I've added some docs and extended the doc strings a bit.

andrewgodwin commented:
Thanks so much! Looks great, and merged in.
